packages/gatsby-theme/src/components/Upvote.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { Heading, Flex } from './shared/base';
+import CustomButton from './CustomButton';
+import { loginUser } from '../utils/auth/auth';
+import WithCurrentUser from '../utils/auth/WithCurrentUser';
+
+// Still to-do:
+// Query the backend with the ID of the tutorial to see whcih tutorial was upvoted
+// Create a way to store which user has upvoted the tutorial so that they can only
+// upvote it once
+
+const Upvote = () => {
+  return (
+    <WithCurrentUser>
+      {({ user }) => {
+        if (user) {
+          return <UpvoteData event={() => console.log('upvoted!')} />;
+        } else {
+          return <UpvoteData event={() => loginUser()} />;
+        }
+      }}
+    </WithCurrentUser>
+  );
+};
+
+type UpvoteDataProps = {
+  event: string;
+};
+
+const UpvoteData: React.FunctionComponent<UpvoteDataProps> = ({ event }) => {
+  return (
+    <Flex flexDirection="column" alignItems="center" justifyContent="center">
+      <CustomButton type="vote" onClick={event} />
```

Comment:
```suggestion
      <VoteButton onClick={event} />
```

It reads nicer if we had a component like this for each type of button. It can be implemented like so:

```jsx
export const VoteButton = (props) => <CustomButton {...props} type="vote" />
```

Very minor edit, but it is easier to read and understand. It also gives us the freedom to change how we implement the button if requirements change in the future.

Reply:
Got it!